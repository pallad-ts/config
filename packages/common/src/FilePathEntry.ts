export type FilePathEntry = string | FilePathEntry.Shape;
export namespace FilePathEntry {
    export interface Shape {
        path: string;
        required: boolean;
    }

    export function normalize(entry: FilePathEntry): FilePathEntry.Shape {
        if (typeof entry === "string") {
            return {
                path: entry,
                required: true,
            };
        }

        return entry;
    }
}
