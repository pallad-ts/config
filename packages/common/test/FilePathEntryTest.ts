import { FilePathEntry } from "@src/FilePathEntry";

describe("FilePathEntry", () => {
    it.each<[FilePathEntry]>([
        ["file.txt"],
        [{ path: "file.txt", required: true }],
        [{ path: "file.txt", required: false }],
    ])("normalization: %s", entry => {
        expect(FilePathEntry.normalize(entry)).toMatchSnapshot();
    });
});
