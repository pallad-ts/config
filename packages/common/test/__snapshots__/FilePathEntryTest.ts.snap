// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilePathEntry normalization: { path: 'file.txt', required: false } 1`] = `
{
  "path": "file.txt",
  "required": false,
}
`;

exports[`FilePathEntry normalization: { path: 'file.txt', required: true } 1`] = `
{
  "path": "file.txt",
  "required": true,
}
`;

exports[`FilePathEntry normalization: file.txt 1`] = `
{
  "path": "file.txt",
  "required": true,
}
`;
